sandbox/visualizations_emp.R

```diff
@@ -11,176 +11,368 @@ library(leaflet.providers)
 
-## compiled image-level metadata. after loading each metadata file, add a row index ("X") identical to the row index in the original data files-- this will allow us to easily merge the two types of files in the next step.
+totalexif6$datetime <- ymd_hms(totalexif6$DateTimeOriginal)
 
-exif_20220630_0041_metadata <- read.csv ("extracted-metadata\\image-level-metadata\\image-metadata_20220630-0041.csv")
+# creating "duration in seconds from beginning of image to collection to time of [i] image" column. the number in brackets is the index position of the image that was collected first.
 
-exif_20220630_0041_metadata$X <- 1:nrow(exif_20220630_0041_metadata)
+totalexif1$duration <- as.duration(totalexif1$datetime[1] %--% totalexif1$datetime)
 
-exif_20220730_0079_metadata <- read.csv ("extracted-metadata\\image-level-metadata\\image-metadata_20220730-0079.csv")
+totalexif2$duration <- as.duration(totalexif2$datetime[1] %--% totalexif2$datetime)
 
-exif_20220730_0079_metadata$X <- 1:nrow(exif_20220730_0079_metadata)
+totalexif3$duration <- as.duration(totalexif3$datetime[1] %--% totalexif3$datetime)
 
-exif_20230528_0008_metadata <- read.csv ("extracted-metadata\\image-level-metadata\\image-metadata_20230528-0008.csv")
+totalexif4$duration <- as.duration(totalexif4$datetime[1] %--% totalexif4$datetime)
 
-exif_20230528_0008_metadata$X <- 1:nrow(exif_20230528_0008_metadata)
+totalexif5$duration <- as.duration(totalexif5$datetime[1] %--% totalexif5$datetime)
 
-exif_20230528_0009_metadata <- read.csv ("extracted-metadata\\image-level-metadata\\image-metadata_20230528-0009.csv")
+totalexif6$duration <- as.duration(totalexif6$datetime[1] %--% totalexif6$datetime)
 
-exif_20230528_0009_metadata$X <- 1:nrow(exif_20230528_0009_metadata)
+#### exif1 interactive map ####
 
-exif_20230706_0152_metadata <- read.csv ("extracted-metadata\\image-level-metadata\\image-metadata_20230706-0152.csv")
+## define functions for continuous coloring
 
-exif_20230706_0152_metadata$X <- 1:nrow(exif_20230706_0152_metadata)
+paletteNumASL1 <- colorNumeric('YlOrRd', domain = totalexif1$altitude_asl)
 
-exif_20230706_0153_metadata <- read.csv ("extracted-metadata\\image-level-metadata\\image-metadata_20230706-0153.csv")
+paletteNumDateTime1 <- colorNumeric('BuPu', domain = totalexif1$duration)
 
-exif_20230706_0153_metadata$X <- 1:nrow(exif_20230706_0153_metadata)
+## the map
 
-## merge original data and image level metadata
+exif1_MAP <-
+  # create leaflet function using data specific to this map
+  leaflet(totalexif1) %>%
+  # use the jawg.terrain basemap
+  addTiles(
+    'https://tile.jawg.io/jawg-terrain/{z}/{x}/{y}{r}.png?access-token=cN5iUzB2bL1QMdeVwMJlflwXQweOoNTDOadGfoCbPf1rRM9FGPb4i1m4f2k5MaJP') %>%
+  # set the center of the map to be the center ish of the flight footprint
+  setView(
+    lng=(totalexif1$lon[888]),
+    lat=(totalexif1$lat[888]),
+    zoom = 16
+    ) %>%
+  # add circles at each image location
+  addCircleMarkers(
```

Comment:
If this layer is not displaying AGL or duration, what is its purpose. I don't understand leaflet well enough to understand why this is necessary.

Reply:
these are just the base circles that exist when both AGL and duration layers are unclicked. they're the same size and in the same place as the other circles. it just means the contents of the map don't entirely pop out of existence when both interactive layers are turned off. I think this is a useful/intuitive feature but I can definitely get rid of these if you want.

Reply:
Got it -- makes sense, thanks.